File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/room/ServiceDatabase.java
package com.timothy.spotifyarchitecture.room;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import com.timothy.spotifyarchitecture.entities.SpotifyUser;
import com.timothy.spotifyarchitecture.entities.Token;

/**
 * Created by tim on 5/30/17.
 */
@Database(entities = {Token.class, SpotifyUser.class}, version = 2)
@TypeConverters({CustomTypeConverter.class})
public abstract class ServiceDatabase extends RoomDatabase {
    public abstract SpotifyDao serviceDao();
}

File: app/src/main/java/com/timothy/musicall/ServiceDatabase.kt
package com.timothy.musicall

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import com.timothy.spotifyarchitecture.entities.SpotifyAlbum
import com.timothy.spotifyarchitecture.entities.SpotifyArtist
import com.timothy.spotifyarchitecture.entities.SpotifyTrack
import com.timothy.spotifyarchitecture.entities.SpotifyUser
import com.timothy.spotifyarchitecture.room.SpotifyConverters
import com.timothy.spotifyarchitecture.room.SpotifyDao
import com.timothy.spotifyarchitecture.room.SpotifyDatabase

/**
 * Room database class
 */
@Database(entities = arrayOf(SpotifyUser::class, SpotifyAlbum::class,
		SpotifyArtist::class, SpotifyTrack::class), version = 3)
@TypeConverters(SpotifyConverters::class)
abstract class ServiceDatabase : RoomDatabase(), SpotifyDatabase {
	override abstract fun spotifyDao(): SpotifyDao
}